src/thread_pool.rs

```diff
@@ -263,13 +268,26 @@ impl WorkerThread {
 
     #[cold]
     pub unsafe fn steal_until(&self, latch: &SpinLatch) {
+        // If another thread stole our job when we panic, we must halt unwinding
+        // until that thread is finished using it.
+        struct PanicGuard<'a>(&'a SpinLatch);
+        impl<'a> Drop for PanicGuard<'a> {
+            fn drop(&mut self) {
+                while !self.0.probe() {
```

Comment:
Hmm, perhaps. scopeguard offers `defer!`, which works in this instance, but not the other case, since we wanted the ability to `mem::forget`...